tiles-generation/cdk/lambda/event-process.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Helper function to process lambda event information.
+ */
+
+const getStateInformation = function(event) {
+    try {
+        let stateInfo;
+        const taskArn = event.resources;
+        const taskLastStatus = event.detail.lastStatus;
+        const container = event.detail.containers[0];
+    
```

Comment:
Too many branches in the code, should we write a switch and make it more modular?

Reply:
Also is there a state diagram that can be referred for this logic?

Reply:
Sure. I will make a diagram for reference.